### ISSUE TITLE
Unable to validate an offering

### ISSUE THREAD

**Issue by michaelneale** [TBD54566975/tbdex-js#251]
Using latest web5/tbdex protocol: 

`Signature verification failed: Expected kid in JWS header to dereference to a DID Document Verification Method`

Versions: 

```
    "@tbdex/http-server": "1.0.2",
    "@web5/common": "1.0.1",
    "@web5/credentials": "1.0.2",
    "@web5/crypto": "1.0.1",
    "@web5/dids": "1.1.0",
```

Test code: 

```typescript

import { DidDht } from '@web5/dids'
import { Offering } from '@tbdex/protocol'


const pfiDid = await DidDht.create({
  options:{

    publish: true,
    services: [{
      id: 'pfi',
      type: 'PFI',
      serviceEndpoint: 'https://example.com/'
    }]
  },
})

const issuerDid = await DidDht.create({
  options:{
    publish: true,
  },
})

const pd = {
  id: 'presentation-definition-kcc',
  name: 'KYC Verification',
  purpose: 'We need to verify your customer status and necessary checks.',
  format: {
    jwt_vc: {
      alg: ['ES256K', 'EdDSA']
    }
  },
  input_descriptors: [
    {
      id: 'known-customer-credential',
      name: 'Known Customer Credential',
      purpose: 'Please present your Known Customer Credential for verification.',
      constraints: {
        fields: [
          {
            path: ['$.type[*]'],
            filter: {
              type: 'string',
              pattern: 'KnownCustomerCredential'
            }
          },
          {
            path: ['$.evidence[*].kind'],
            filter: {
              type: 'string',
              pattern: 'sanction_screening'
            }
          },
          {
            path: ['$.credentialSubject.countryOfResidence'],
            filter: {
              type: 'string',
              const: 'US'
            }
          },
          {
            path: ['$.issuer'],
            filter: {
              type: 'string',
              const: issuerDid.uri
            }
          }
        ]
      }
    }
  ]
}


const offering = Offering.create({
  metadata: {
    from: pfiDid.uri,
    protocol: '1.0'
  },
  data: {
    description: 'Selling BTC for USD',
    payin: {
      currencyCode: 'USD',
      methods: [{
        kind: 'DEBIT_CARD',
        requiredPaymentDetails: {
          '$schema': 'http://json-schema.org/draft-07/schema',
          'type': 'object',
          'properties': {
            'cardNumber': {
              'type': 'string',
              'description': 'The 16-digit debit card number',
              'minLength': 16,
              'maxLength': 16
            },
            'expiryDate': {
              'type': 'string',
              'description': 'The expiry date of the card in MM/YY format',
              'pattern': '^(0[1-9]|1[0-2])\\/([0-9]{2})$'
            },
            'cardHolderName': {
              'type': 'string',
              'description': 'Name of the cardholder as it appears on the card'
            },
            'cvv': {
              'type': 'string',
              'description': 'The 3-digit CVV code',
              'minLength': 3,
              'maxLength': 3
            }
          }
        }
      }]
    },
    payout: {
      currencyCode: 'BTC',
      methods: [
        {
          kind: 'BTC_ADDRESS',
          estimatedSettlementTime: 60,
        }
      ]
    },
    payoutUnitsPerPayinUnit: '0.00003826',
    requiredClaims: pd
  }
})

await offering.sign(pfiDid)

console.log(offering)

await offering.verifySignature()
```

**Comment by nitro-neal** [comment]
latest dids:
https://www.npmjs.com/package/@web5/dids/v/0.0.0-next-16eb49d-20240612013538

and creds:
https://www.npmjs.com/package/@web5/credentials/v/0.0.0-next-16eb49d-20240612013538

**Comment by nitro-neal** [comment]
similar test on mainline that passes - https://github.com/TBD54566975/web5-js/blob/main/packages/credentials/tests/jwt.spec.ts#L76

yea seems like it does not work in latest versions... Hmm will keep investigating..

**Comment by nitro-neal** [comment]
```
// UNIT TEST IN MAIN BRANCH


      const did = await DidDht.create();
      console.log(JSON.stringify(did.document, null, 2))

      const dereferenceResult = await Jwt.didResolver.dereference(did.uri);
      console.log(JSON.stringify(dereferenceResult, null, 2))
      
Output:
{
  "id": "did:dht:piuuzeb3y6j7cg49bc4bw3f67nkocjhqdo7njktj6thoxyua518y",
  "verificationMethod": [
    {
      "id": "did:dht:piuuzeb3y6j7cg49bc4bw3f67nkocjhqdo7njktj6thoxyua518y#0",
      "type": "JsonWebKey",
      "controller": "did:dht:piuuzeb3y6j7cg49bc4bw3f67nkocjhqdo7njktj6thoxyua518y",
      "publicKeyJwk": {
        "crv": "Ed25519",
        "kty": "OKP",
        "x": "bWc7oDkHk9YbXws0GmS-6JUGJ44cOiSqKfR5B4J43I4",
        "kid": "6cX1kEsnP3rZvFkI4tyRUPEkZpFxq7NkWG13iF0LJ0E",
        "alg": "EdDSA"
      }
    }
  ],
  "authentication": [
    "did:dht:piuuzeb3y6j7cg49bc4bw3f67nkocjhqdo7njktj6thoxyua518y#0"
  ],
  "assertionMethod": [
    "did:dht:piuuzeb3y6j7cg49bc4bw3f67nkocjhqdo7njktj6thoxyua518y#0"
  ],
  "capabilityDelegation": [
    "did:dht:piuuzeb3y6j7cg49bc4bw3f67nkocjhqdo7njktj6thoxyua518y#0"
  ],
  "capabilityInvocation": [
    "did:dht:piuuzeb3y6j7cg49bc4bw3f67nkocjhqdo7njktj6thoxyua518y#0"
  ]
}

{
   "dereferencingMetadata":{
      "contentType":"application/did+json"
   },
   "contentStream":{
      "id":"did:dht:piuuzeb3y6j7cg49bc4bw3f67nkocjhqdo7njktj6thoxyua518y",
      "verificationMethod":[
         {
            "id":"did:dht:piuuzeb3y6j7cg49bc4bw3f67nkocjhqdo7njktj6thoxyua518y#0",
            "type":"JsonWebKey",
            "controller":"did:dht:piuuzeb3y6j7cg49bc4bw3f67nkocjhqdo7njktj6thoxyua518y",
            "publicKeyJwk":{
               "kty":"OKP",
               "crv":"Ed25519",
               "x":"bWc7oDkHk9YbXws0GmS-6JUGJ44cOiSqKfR5B4J43I4",
               "kid":"6cX1kEsnP3rZvFkI4tyRUPEkZpFxq7NkWG13iF0LJ0E",
               "alg":"EdDSA"
            }
         }
      ],
      "authentication":[
         "did:dht:piuuzeb3y6j7cg49bc4bw3f67nkocjhqdo7njktj6thoxyua518y#0"
      ],
      "assertionMethod":[
         "did:dht:piuuzeb3y6j7cg49bc4bw3f67nkocjhqdo7njktj6thoxyua518y#0"
      ],
      "capabilityDelegation":[
         "did:dht:piuuzeb3y6j7cg49bc4bw3f67nkocjhqdo7njktj6thoxyua518y#0"
      ],
      "capabilityInvocation":[
         "did:dht:piuuzeb3y6j7cg49bc4bw3f67nkocjhqdo7njktj6thoxyua518y#0"
      ]
   },
   "contentMetadata":{
      "published":true,
      "versionId":"1718218534"
   }
}

```


```
Using latest versions in sample project
import { DidDht, utils as didUtils } from '@web5/dids';

import {Jwt} from '@web5/credentials'

const did = await DidDht.create();
console.log(JSON.stringify(did.document, null, 2))

const dereferenceResult = await Jwt.didResolver.dereference(did.uri);
console.log("~~~dereferenceResult", JSON.stringify(dereferenceResult))

{
  "id": "did:dht:9odrgmsppx3s53qyhg7xj7ucfbt1dxbkoo7ixera9cd74d9d8nxo",
  "verificationMethod": [
    {
      "id": "did:dht:9odrgmsppx3s53qyhg7xj7ucfbt1dxbkoo7ixera9cd74d9d8nxo#0",
      "type": "JsonWebKey",
      "controller": "did:dht:9odrgmsppx3s53qyhg7xj7ucfbt1dxbkoo7ixera9cd74d9d8nxo",
      "publicKeyJwk": {
        "crv": "Ed25519",
        "kty": "OKP",
        "x": "_AZDLs1r823lwOG69PZsKGMhvCqEO1egmPsH3Q_jOJ8",
        "kid": "TTbrK7q85QHRO82qUsEiuqo1NaqFA8I5_KseO6yFESE",
        "alg": "EdDSA"
      }
    }
  ],
  "authentication": [
    "did:dht:9odrgmsppx3s53qyhg7xj7ucfbt1dxbkoo7ixera9cd74d9d8nxo#0"
  ],
  "assertionMethod": [
    "did:dht:9odrgmsppx3s53qyhg7xj7ucfbt1dxbkoo7ixera9cd74d9d8nxo#0"
  ],
  "capabilityDelegation": [
    "did:dht:9odrgmsppx3s53qyhg7xj7ucfbt1dxbkoo7ixera9cd74d9d8nxo#0"
  ],
  "capabilityInvocation": [
    "did:dht:9odrgmsppx3s53qyhg7xj7ucfbt1dxbkoo7ixera9cd74d9d8nxo#0"
  ]
}

{
   "dereferencingMetadata":{
      "contentType":"application/did+json"
   },
   "contentStream":{
      "id":"did:dht:9odrgmsppx3s53qyhg7xj7ucfbt1dxbkoo7ixera9cd74d9d8nxo",
      "verificationMethod":[
         {
            "id":"did:dht:9odrgmsppx3s53qyhg7xj7ucfbt1dxbkoo7ixera9cd74d9d8nxo#0",
            "type":"JsonWebKey",
            "controller":"did:dht:9odrgmsppx3s53qyhg7xj7ucfbt1dxbkoo7ixera9cd74d9d8nxo",
            "publicKeyJwk":{
               "kty":"OKP",
               "crv":"Ed25519",
               "x":"_AZDLs1r823lwOG69PZsKGMhvCqEO1egmPsH3Q_jOJ8",
               "kid":"TTbrK7q85QHRO82qUsEiuqo1NaqFA8I5_KseO6yFESE",
               "alg":"Ed25519"
            }
         }
      ],
      "authentication":[
         "did:dht:9odrgmsppx3s53qyhg7xj7ucfbt1dxbkoo7ixera9cd74d9d8nxo#0"
      ],
      "assertionMethod":[
         "did:dht:9odrgmsppx3s53qyhg7xj7ucfbt1dxbkoo7ixera9cd74d9d8nxo#0"
      ],
      "capabilityDelegation":[
         "did:dht:9odrgmsppx3s53qyhg7xj7ucfbt1dxbkoo7ixera9cd74d9d8nxo#0"
      ],
      "capabilityInvocation":[
         "did:dht:9odrgmsppx3s53qyhg7xj7ucfbt1dxbkoo7ixera9cd74d9d8nxo#0"
      ]
   },
   "contentMetadata":{
      "published":true,
      "versionId":"1718218620"
   }
}

```

Notice the `alg` in the dereferencingMetadata dereferenceResult is Ed25519 and not the correct value of EdDSA. This is what is causing the problem, in mainline the resolver is giving the right thing, but in our `next` versions it is giving another,

**Comment by nitro-neal** [comment]
ok! I cut a new release and this is working in my sample project:

package.json
```
{
  "name": "web5-js-test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "type": "module",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@web5/api": "^0.9.4",
    "@web5/credentials": "^1.0.3",
    "@web5/crypto": "^1.0.1",
    "@web5/dids": "^1.1.1"
  }
}

```

index.js
```
import { DidDht, utils as didUtils } from '@web5/dids';

import {Jwt} from '@web5/credentials'

const did = await DidDht.create();
console.log("DidDocument:")
console.log(JSON.stringify(did.document, null, 2))

const dereferenceResult = await Jwt.didResolver.dereference(did.uri);
console.log("Resolver Result:");
console.log(JSON.stringify(dereferenceResult))

const siopv2Response = {
  id_token: await Jwt.sign({
    signerDid : did,
    payload   : {
      iss   : did.uri,
      sub   : did.uri,
      aud   : 'did:dht:ho3axp5pgp4k8a7kqtb8knn5uaqwy9ghkm98wrytnh67bsn7ezry',
      nonce : 'd844f80d21c33ea6e087afa2b84dc31f',
      iat   : Math.floor(Date.now() / 1000),
      exp   : Math.floor(Date.now() / 1000) + (30 * 60), // plus 30 minutes
    }
  })
};

// Verify the JWT and make sure we get a result that it does not throw an error.
const jwtVerifyResult = await Jwt.verify({ jwt: siopv2Response.id_token });

console.log("JWT Verification Result:");
console.log(JSON.stringify(jwtVerifyResult, null, 2))
```


output:
```
❯ node index.js
DidDocument:
{
  "id": "did:dht:ircwikw63j4mrx6as5ky3xpz4jjeza8tpwaxomenq3egfh3zifio",
  "verificationMethod": [
    {
      "id": "did:dht:ircwikw63j4mrx6as5ky3xpz4jjeza8tpwaxomenq3egfh3zifio#0",
      "type": "JsonWebKey",
      "controller": "did:dht:ircwikw63j4mrx6as5ky3xpz4jjeza8tpwaxomenq3egfh3zifio",
      "publicKeyJwk": {
        "crv": "Ed25519",
        "kty": "OKP",
        "x": "qRlKqp7KdLI_2LbUDL230lKL4PFtMPgtAnZQYvM3qWs",
        "kid": "sBRgIsXVlrabv7YZ00XmAvMyUd-d2fP0Dh7aD_hi0oQ",
        "alg": "EdDSA"
      }
    }
  ],
  "authentication": [
    "did:dht:ircwikw63j4mrx6as5ky3xpz4jjeza8tpwaxomenq3egfh3zifio#0"
  ],
  "assertionMethod": [
    "did:dht:ircwikw63j4mrx6as5ky3xpz4jjeza8tpwaxomenq3egfh3zifio#0"
  ],
  "capabilityDelegation": [
    "did:dht:ircwikw63j4mrx6as5ky3xpz4jjeza8tpwaxomenq3egfh3zifio#0"
  ],
  "capabilityInvocation": [
    "did:dht:ircwikw63j4mrx6as5ky3xpz4jjeza8tpwaxomenq3egfh3zifio#0"
  ]
}
Resolver Result:
{"dereferencingMetadata":{"contentType":"application/did+json"},"contentStream":{"id":"did:dht:ircwikw63j4mrx6as5ky3xpz4jjeza8tpwaxomenq3egfh3zifio","verificationMethod":[{"id":"did:dht:ircwikw63j4mrx6as5ky3xpz4jjeza8tpwaxomenq3egfh3zifio#0","type":"JsonWebKey","controller":"did:dht:ircwikw63j4mrx6as5ky3xpz4jjeza8tpwaxomenq3egfh3zifio","publicKeyJwk":{"kty":"OKP","crv":"Ed25519","x":"qRlKqp7KdLI_2LbUDL230lKL4PFtMPgtAnZQYvM3qWs","kid":"sBRgIsXVlrabv7YZ00XmAvMyUd-d2fP0Dh7aD_hi0oQ","alg":"EdDSA"}}],"authentication":["did:dht:ircwikw63j4mrx6as5ky3xpz4jjeza8tpwaxomenq3egfh3zifio#0"],"assertionMethod":["did:dht:ircwikw63j4mrx6as5ky3xpz4jjeza8tpwaxomenq3egfh3zifio#0"],"capabilityDelegation":["did:dht:ircwikw63j4mrx6as5ky3xpz4jjeza8tpwaxomenq3egfh3zifio#0"],"capabilityInvocation":["did:dht:ircwikw63j4mrx6as5ky3xpz4jjeza8tpwaxomenq3egfh3zifio#0"]},"contentMetadata":{"published":true,"versionId":"1718220160"}}
JWT Verification Result:
{
  "header": {
    "typ": "JWT",
    "alg": "EdDSA",
    "kid": "did:dht:ircwikw63j4mrx6as5ky3xpz4jjeza8tpwaxomenq3egfh3zifio#0"
  },
  "payload": {
    "iss": "did:dht:ircwikw63j4mrx6as5ky3xpz4jjeza8tpwaxomenq3egfh3zifio",
    "sub": "did:dht:ircwikw63j4mrx6as5ky3xpz4jjeza8tpwaxomenq3egfh3zifio",
    "aud": "did:dht:ho3axp5pgp4k8a7kqtb8knn5uaqwy9ghkm98wrytnh67bsn7ezry",
    "nonce": "d844f80d21c33ea6e087afa2b84dc31f",
    "iat": 1718220160,
    "exp": 1718221960
  }
}
```

**Comment by michaelneale** [comment]
thanks @nitro-neal confirmed it works - but we will need a new tbdex to refer to this
